Generator/Sources/NeedleFramework/Parsing/Pluginized/Processors/NonCoreComponentLinker.swift

```diff
@@ -39,13 +39,19 @@ class NonCoreComponentLinker: Processor {
         var nonCoreMap = [String: ASTComponent]()
         for nonCoreComponent in nonCoreComponents {
             nonCoreMap[nonCoreComponent.name] = nonCoreComponent
+            if nonCoreComponent.isLeaf {
+                nonCoreComponent.isLeaf = false
+            }
```

Comment:
It wasn't clear to me what this was accomplishing. How would `isLeaf` be set incorrectly here? Perhaps a comment explaining the situation would be helpful.

Reply:
Added a comment